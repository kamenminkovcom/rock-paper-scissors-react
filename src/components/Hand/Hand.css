.rotate {
    transform: scaleX(-1);
    filter: FlipH;
}

.hand {
    background-size: contain;
    background: no-repeat center;
    width: 30%;
}