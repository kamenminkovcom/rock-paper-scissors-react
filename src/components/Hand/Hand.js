import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './Hand.css';

const Hand = ({isCpu, handType}) => (
    <div className={classNames('hand',
        'background-no-repeat',
        'background-contain-size',
        {rotate: isCpu},
        handType)}/>
);

Hand.propTypes = {
    isCpu: PropTypes.bool,
    handType: PropTypes.string.isRequired
};

export default Hand;
