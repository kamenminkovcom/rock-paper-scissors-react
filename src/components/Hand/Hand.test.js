import React from 'react';
import Hand from './Hand';
import {shallow} from 'enzyme';

describe('<Hand/>', () => {
    const handType = 'rock';
    const isCpu = true;
    const wrapper = shallow(<Hand isCpu={true} handType={handType}/>);
    const classNames = wrapper.props().className.split(' ');

    it('Should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Should contain handType className', () => {
        expect(classNames.includes(handType)).toEqual(true);
    });

    it('Should contain rotate className', () => {
        const reverseClassName = 'rotate';
        expect(classNames.includes(reverseClassName)).toEqual(true);
    });
});