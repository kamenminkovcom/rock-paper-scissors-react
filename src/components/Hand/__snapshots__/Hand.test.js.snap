// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hand/> Should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Hand
    handType="rock"
    isCpu={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "hand background-no-repeat background-contain-size rotate rock",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "hand background-no-repeat background-contain-size rotate rock",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
